### PR TITLE
8321802: (zipfs) Add validation of incorrect LOC signature in ZipFileSystem

### PR DESCRIPTION
**Pull request by eirbjo**
Please review this PR which adds validation of incorrect LOC signatures in `ZipFileSystem`.

`ZipFile` already rejects the case where the  offset pointed to from the CEN header does not start with the expected LOC signature. It makes sense to add this check to `ZipFileSystem` as well.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8321802](https://bugs.openjdk.org/browse/JDK-8321802): (zipfs) Add validation of incorrect LOC signature in ZipFileSystem (**Enhancement** - P4)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17059/head:pull/17059` \
`$ git checkout pull/17059`

Update a local copy of the PR: \
`$ git checkout pull/17059` \
`$ git pull https://git.openjdk.org/jdk.git pull/17059/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17059`

View PR using the GUI difftool: \
`$ git pr show -t 17059`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17059.diff">https://git.openjdk.org/jdk/pull/17059.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17059#issuecomment-1851692069)

### CODE REVIEW

**Review by AlanBateman — APPROVED** [review]

File: src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java [hunk 1]
```diff
@@ -2575,6 +2575,9 @@ private void initDataPos() throws IOException {
                 if (readNBytesAt(buf, 0, buf.length, pos) != LOCHDR) {
                     throw new ZipException("invalid loc " + pos + " for entry reading");
                 }
+                if (LOCSIG(buf) != LOCSIG) {
+                    throw new ZipException("invalid loc header (bad signature)");
```
Comment (LanceAndersen) [inline src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java:2579]: Please change loc -> LOC

Comment (eirbjo) [reply src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java]: Fixed, including the ZipException for the preceding read check.

**Review by LanceAndersen — APPROVED** [review]

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**